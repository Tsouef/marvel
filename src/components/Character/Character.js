import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchCharacterById } from '../../actions';
import './Character.css';

class Character extends Component {
  componentDidMount() {
    this.props.fetchCharacterById(this.props.match.params.id);
  }

  render() {
    if (this.props.characters.character.length <= 0) {
      return <div>Waiting</div>;
    }

    const { thumbnail, name, description } = this.props.characters.character;

    return (
      <div className="character">
        <div className="character-cover">
          <img alt="" src={`${thumbnail.path}.${thumbnail.extension}`} />
        </div>
        <div className="character-description">
          <p>
            <span className="meta">Name: </span> {name}
          </p>
          <p>
            <span className="meta">Description: </span>
            <span dangerouslySetInnerHTML={{ __html: description }} />
          </p>
        </div>
      </div>
    );
  }
}

function mapStateToProps({ characters }) {
  return { characters };
}

export default connect(mapStateToProps, { fetchCharacterById })(Character);
