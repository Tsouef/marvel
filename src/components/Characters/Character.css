.character {
  margin: 20px 0 0 0;
  display: flex;
  justify-content: space-between;
  flex-direction: column;
  flex-wrap: nowrap;
}

.character-description {
  display: flex;
  flex-direction: column;
  margin: 0 0 0 20px;
}

.meta {
  font-weight: bold;
  font-size: 15px;
}

.character-characters li {
  margin: 0 0 0 15px;
  line-height: 1.4;
}
