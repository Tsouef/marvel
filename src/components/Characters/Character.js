import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchCharacterById } from '../../actions';
import './Character.css';

export class Character extends Component {
  componentDidMount() {
    if (this.props.match) {
      this.props.fetchCharacterById(this.props.match.params.id);
    }
  }

  render() {
    if (
      this.props.characters.isFetching ||
      this.props.characters.character.length <= 0
    ) {
      return <div>Loading...</div>;
    }

    const { thumbnail, name, description } = this.props.characters.character;

    return (
      <div className="character">
        <div className="character-cover">
          <img alt="" src={`${thumbnail.path}.${thumbnail.extension}`} />
        </div>
        <div className="character-description">
          <p>
            <span className="meta">Name: </span> {name}
          </p>
          <p>
            <span className="meta">Description: </span>
            <span dangerouslySetInnerHTML={{ __html: description }} />
          </p>
        </div>
      </div>
    );
  }
}

function mapStateToProps({ characters }) {
  return { characters };
}

export default connect(mapStateToProps, { fetchCharacterById })(Character);
