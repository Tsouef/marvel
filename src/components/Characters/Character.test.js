import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Character } from './Character';

describe('Character Component', () => {
  it('should render Character', () => {
    const props = {
      characters: {
        character: [],
        isFetching: false
      }
    };

    const tree = renderer
      .create(
        <MemoryRouter>
          <Character {...props} />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with props', () => {
    // const props = {
    //
    // };
  });

  // it('should render Character with elements', () => {
  //   const props = {
  //     favorites: {
  //       favorites: [
  //         {
  //           thumbnail: 'thumbnail',
  //           name: 'name',
  //           description: 'description',
  //           id: 1
  //         }
  //       ]
  //     }
  //   };
  //
  //   const tree = renderer.create(<Character {...props} />).toJSON();
  //   expect(tree).toMatchSnapshot();
  // });
});
