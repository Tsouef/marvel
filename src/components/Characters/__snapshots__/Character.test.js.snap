// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Component should render Character 1`] = `
<div>
  Loading...
</div>
`;

exports[`Character Component should render with props 1`] = `
<div
  className="character"
>
  <div
    className="character-cover"
  >
    <img
      alt=""
      src="path.jpg"
    />
  </div>
  <div
    className="character-description"
  >
    <p>
      <span
        className="meta"
      >
        Name: 
      </span>
       
      character
    </p>
    <p>
      <span
        className="meta"
      >
        Description: 
      </span>
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "text",
          }
        }
      />
    </p>
  </div>
</div>
`;
